core/src/io/anuke/mindustry/ui/dialogs/CustomGameDialog.java

```diff
@@ -155,4 +158,89 @@ private void displayGameModeHelp(){
         d.show();
     }
 
-}
+    private class CustomRulesDialog extends FloatingDialog{
```

Comment:
This should not be an inner class. It's may be related, but it's still a separate dialog and a rather large class file.

Reply:
How it should be related to make it consistent? Should I make a function that will sync vars between classes, or in ``CustomRulesDialog`` should I just access vars by ``ui.custom.``?